### ISSUE TITLE
OS X: build break with AppleClang 8.0.0

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#7275]
Seen after installing a very recent Xcode.
```
Invoking "/Users/russellk/git/coreclr/src/pal/tools/gen-buildsys-clang.sh" "/Users/russellk/git/coreclr" 3 5 x64 Checked  Include_Tests
Including tests directory in build.
-- The C compiler identification is AppleClang 8.0.0.8000038
-- The CXX compiler identification is AppleClang 8.0.0.8000038
```
Error details
```
In file included from /Users/russellk/git/coreclr/src/pal/src/sharedmemory/sharedmemory.cpp:9:
/Users/russellk/git/coreclr/src/pal/src/include/pal/malloc.hpp:134:13: error: destructor called on non-final 'SharedMemoryProcessDataBase' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
            p->~T();
            ^
/Users/russellk/git/coreclr/src/pal/src/sharedmemory/sharedmemory.cpp:879:13: note: in instantiation of function template specialization 'CorUnix::InternalDelete<SharedMemoryProcessDataBase>' requested here
            InternalDelete(m_data);
            ^
/Users/russellk/git/coreclr/src/pal/src/include/pal/malloc.hpp:134:17: note: qualify call to silence this warning
            p->~T();
                ^
                SharedMemoryProcessDataBase::
1 error generated.
```

**Comment by gkhanna79** [comment]
@JohnChen0 can help take a look at this.

**Comment by janvorli** [comment]
The fix would be simple - adding empty virtual destructor to the SharedMemoryProcessDataBase.

**Comment by RussKeldorph** [comment]
@mmitche FYI

**Comment by JohnChen0** [comment]
Resolved with #7292.
